lti_store/key_handlers.py

```diff
@@ -0,0 +1,282 @@
+                # as an invalid key when the JWK loading fails.
+                raise NoSuitableKeys() from err
+            keyset.extend(keys)
+
+        if self.public_key and kid:
+            # Fill in key id of stored key.
+            # This is needed because if the JWS is signed with a
+            # key with a kid, pyjwkest doesn't match them with
+            # keys without kid (kid=None) and fails verification
+            self.public_key.kid = kid
+
+            # Add to keyset.
+            keyset.append(self.public_key)
+
+        return keyset
+
+    def validate_and_decode(self, token):
+        """Check if a message sent by the tool is valid.
+
+        The authorization server decodes the JWT and MUST validate the values for the
+        iss, sub, exp, aud and jti claims.
+
+        Args:
+            token (str): JWT Token.
+
+        Raises:
+            TokenSignatureExpired: JWT token signature is expired.
+            NoSuitableKeys: JWKS could not be loaded.
+            MalformedJwtToken: JWT token is malformed.
+            BadJwtSignature: JWT token signature is invalid.
+
+        References:
+            https://www.imsglobal.org/spec/security/v1p0/#using-oauth-2-0-client-credentials-grant
+
+        """
+        try:
+            # Get KID from JWT header.
+            jwt = JWT().unpack(token)
+
+            # Verify message signature.
+            message = JWS().verify_compact(
+                token, keys=self._get_keyset(jwt.headers.get("kid"))
+            )
+
+            # If message is valid, check expiration from JWT.
+            if "exp" in message and message["exp"] < time.time():
+                raise TokenSignatureExpired()
+
+            # Return decoded message.
+            return message
+
+        except NoSuitableSigningKeys as err:
+            raise NoSuitableKeys() from err
+        except (BadSyntax, WrongNumberOfParts) as err:
+            raise MalformedJwtToken() from err
+        except BadSignature as err:
+            raise BadJwtSignature() from err
+
+
+class PlatformKeyHandler:
+    """Platform RSA Key handler.
+
+    This class loads the platform key and is responsible for
+    encoding JWT messages and exporting public keys.
+
+    Attributes:
+        key (:obj:`str`, optional): RSA Key.
+
+    """
+
+    def __init__(self, key_pem, kid=None):
+        """Import key when instancing class if a key is present.
+
+        Args:
+            key_pem (str): Key PEM.
+            kid (:obj:`str`, optional): Private Key ID.
+
+        Raises:
+            InvalidRsaKey: Failed to import key.
+
+        """
+        self.key = None
+
+        if key_pem:
+            # Import JWK from RSA key.
+            try:
+                self.key = RSAKey(
+                    kid=kid,
+                    key=RSA.import_key(key_pem),
+                )
+            except ValueError as err:
+                raise InvalidRsaKey() from err
```

Comment:
```suggestion
                raise InvalidRsaKey()
```
Is this statement required? @kuipumu

Reply:
It's not required but makes the exception chain explicit, should I also make it implicit on all other exceptions?